protos/flyteidl/core/types.proto

```diff
@@ -64,6 +64,11 @@ message EnumType {
     repeated string values = 1;
 }
 
+message TypeAnnotation {
```

Comment:
```suggestion
// TypeAnnotation encapsulates registration time information about a type. This can be used for various control-plane operations. TypeAnnotation will not be available at runtime when a task runs.
message TypeAnnotation {
```